### PR TITLE
F*: core lib: misc. improvements

### PR DESCRIPTION
**Pull request by W95Psp**
This PR reflects the additional definitions, properties, and lemma we added while working on Kyber verification.

This PR also introduces issues #422 and #423.

### CODE REVIEW

File: proof-libs/fstar/rust_primitives/Rust_primitives.Arrays.fsti
Comment (franziskuskiefer) [inline proof-libs/fstar/rust_primitives/Rust_primitives.Arrays.fsti:1]: This, and most of the others need comments.

File: proof-libs/fstar/rust_primitives/Rust_primitives.BitVectors.fsti [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+           | 15  -> Some 4
+           | 31  -> Some 5
+           | 63  -> Some 6
+           | 127 -> Some 7
+           | 255 -> Some 8
+           | 511 -> Some 9
+           | 1023  -> Some  10
+           | 2047  -> Some  11
+           | 4095  -> Some  12
+(*           | 8191  -> Some  13
```
Comment (franziskuskiefer) [inline proof-libs/fstar/rust_primitives/Rust_primitives.BitVectors.fsti:71]: Remove?

**Review by franziskuskiefer — COMMENTED** [review]
I didn't check it all in detail. This should be the last PR like this.
But please add some comments.

File: proof-libs/fstar/rust_primitives/Rust_primitives.BitVectors.fsti (hunk 1 above)
Comment (W95Psp) [reply proof-libs/fstar/rust_primitives/Rust_primitives.BitVectors.fsti]: Fixed!

**Review by franziskuskiefer — APPROVED** [review]

**Pull request merged**